tailscale/resource_dns_split_nameservers.go

```diff
@@ -59,47 +60,39 @@ func resourceSplitDNSNameserversRead(ctx context.Context, d *schema.ResourceData
 	return nil
 }
 
-func resourceSplitDNSNameserversCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
-	client := m.(*Clients).V1
+func resourceSplitDNSNameserversCreateOrUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*Clients).V2
 	nameserversSet := d.Get("nameservers").(*schema.Set)
 	domain := d.Get("domain").(string)
 
 	nameserversList := nameserversSet.List()
 
-	req := make(tailscale.SplitDnsRequest)
+	req := make(tsclient.SplitDNSRequest)
 	var nameservers []string
 	for _, nameserver := range nameserversList {
 		nameservers = append(nameservers, nameserver.(string))
 	}
 	req[domain] = nameservers
 
 	// Return value is not useful to us here, ignore.
-	if _, err := client.UpdateSplitDNS(ctx, req); err != nil {
-		return diagnosticsError(err, "Failed to set dns split nameservers")
+	if _, err := client.DNS().UpdateSplitDNS(ctx, req); err != nil {
+		return diagnosticsError(err, "Failed to set create dns split nameservers")
```

Comment:
Was this error line intentionally changed?

Reply:
I've recently taken to differentiating between create and update flows. Of course, in this case, they're the same code. Will change back.

Reply:
Oh, the message had both create and set, oops :)